theano/tensor/nnet/neighbours.py

```diff
@@ -80,11 +80,14 @@ def make_node(self, ten4, neib_shape, neib_step=None):
             neib_step = neib_shape
         else:
             neib_step = T.as_tensor_variable(neib_step)
-
+        
         assert ten4.ndim == 4
         assert neib_shape.ndim == 1
         assert neib_step.ndim == 1
 
+        self.neib_shape = neib_shape
+        self.neib_step = neib_step
```

Comment:
This is not good. You can't keep in the op a link to the Theano variables for the Apply. The same op is shared between many apply. There would be collision. Remove that change. In the infer_shape, you can access them throught the node parameter.